drivers/ieee802154/ieee802154_rf2xx.c

```diff
@@ -211,6 +211,13 @@ static void rf2xx_thread_main(void *arg)
 			k_timer_start(&ctx->trx_isr_timeout, K_MSEC(10), 0);
 		} else if (isr_status & (1 << RF2XX_TRX_END)) {
 			if (ctx->trx_state == RF2XX_TRX_PHY_BUSY_RX) {
+				/* Ensures that automatically ACK will be sent
+				 * when requested
+				 */
+				while (rf2xx_iface_reg_read(dev,
```

Comment:
maybe move that logic into a dedicated function (with a relevant name about auto ack reply).
Are you entirely sure it's safe to loop without any counter to guard it case equality never happens? (hw issue, etc..)

Reply:
You can have a very good overview about what happen looking at **Figure 7-12. Flow Diagram of RX_AACK** on the AT86RF233 datasheet.
![rx_fsm](https://user-images.githubusercontent.com/7027887/71979575-d0a1b680-31fc-11ea-8882-607c4a531920.jpg)

The issue currently happen because the driver switch to PLL_ON and this can aborts the flow after TRX_END depending how big is the frame and how fast system reacts. 

The driver currently works only on the gray part of this flow which is processed mostly by BUSY_RX_AACK state in the transceiver. We have a timeout to cover frames that don't generate valid FCS. After IRQ3 (TRX_END) we need ensures that transceiver **is not on BUSY_RX_AACK state to move forward**. The FSM transition from BUSY_RX_AACK to RX_AACK_ON is automatically generated by the transceiver without any intervention.

We could maybe check `ctx->trx_state != RF2XX_TRX_PHY_STATE_IDLE` condition on the loop as an alternative to counter. This will ensures that the timeout could break the loop. It is platform independent and we already had it implemented on the flow.

What do you think?

Reply:
sounds fine yes